enterprise/views.py

```diff
@@ -656,9 +733,15 @@ def get(self, request, enterprise_uuid, course_id):
 
         enterprise_customer, course, modes = self.get_base_details(enterprise_uuid, course_id)
 
+        # Verify that the request user belongs to the enterprise against the
+        # provided `enterprise_uuid` and if user does not belong to the
+        # enterprise then redirect the user to the ecommerce flow in LMS.
+        enterprise_customer_user = get_enterprise_customer_user(request.user.id, enterprise_uuid)
```

Comment:
The right flow for this is to [link the leaner with the given enterprise if they are not already linked and show them the course enrollment landing page](https://openedx.atlassian.net/browse/ENT-379?focusedCommentId=266062&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-266062). I will update this.